src/uct/base/uct_iface.c

```diff
@@ -136,7 +136,52 @@ ucs_status_t uct_iface_get_address(uct_iface_h iface, uct_iface_addr_t *addr)
 int uct_iface_is_reachable(uct_iface_h iface, const uct_iface_addr_t *addr)
 {
     return iface->ops.iface_is_reachable(iface, addr);
+}
+
+ucs_status_t uct_wakeup_open(uct_iface_h iface, unsigned events,
+                             uct_wakeup_h *wakeup_p)
+{
+    if ((events == 0) || (wakeup_p == NULL)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    *wakeup_p = ucs_malloc(sizeof(**wakeup_p), "iface_wakeup_context");
+    if (*wakeup_p == NULL) {
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    (*wakeup_p)->fd = -1;
+    (*wakeup_p)->iface = iface;
+    (*wakeup_p)->events = events;
+
+    return iface->ops.iface_wakeup_open(iface, events, wakeup_p);
```

Comment:
why are you passing pointer-to-pointer if you are not returning a pointer from the called function?